### ISSUE TITLE
Search doesn't respect filter (filter field = 'x' returns field = 'y')

### ISSUE THREAD

**Issue by flexchar** [meilisearch/meilisearch#3257]
Hi! First of all, thanks for amazing piece of software. Been a big fan of what you guys are doing but recently something weird began happening. We have search that needs certain constraints to be applied but they're ignored.

We're using Laravel Scout/Meilisearch-php library but I was able to isolate using just postman and API endpoint. 

Meilisearch Server: 0.29.1
Environment: Development (so we can have dashboard available)

Masterkey is set, no errors during boot. It is deployed on internal network only kubernetes environment. I don't think that's relevant. 

Endpoint: POST `{{url}}/indexes/{{indexUID}}/search`

Request:
```json
{
    "q": "NCS",
    "filter": "standard = true AND product_filter = 'ab_cd nice 10'"
}
```

Response: 

```json
{
    "hits": [
        {
            "frontend_name": "NCS S 1002-Y50R",
            "base_color_name": "Hvid",
            "product_filter": "ab_123",
            "standard": false,
        }
    ],
    "estimatedTotalHits": 1,
    "query": "NCS",
    "limit": 20,
    "offset": 0,
    "processingTimeMs": 0
}
```

> I've replaces the product filter in the request and in the response with random strings that match exactly the same length. 

This result shall NOT be returned but it comes up. I'm clueless and worried that meilisearch will not respect filters in other projects - scenario such as the patient records described in Multitenancy documentation. 

The same happens when `q=` is empty, and when searching for this record without `standard` filter set (so `filter": "product_filter = 'fg_yu lack 50'"`). 

Before populating the index has the following settings applied:

```php
// updateSortableAttributes
'frontend_name',
'base_color_name',

// updateFilterableAttributes
'base_color_name',
'product_filter',
'standard',

// updateSortableAttributes
'id',
```

I omitted certain fields from examples that seems to not be relevant but order of fields are retained. 

I'm planning on updating however the need for rebuilding index and demand for usage is preventing. Index incompatibility after updating is probably the biggest pain point on a side note.

**Comment by oluademola** [comment]
Hello @flexchar,

We are sorry for the inconvenience. Unfortunately, updating your Meilisearch server is the only way out of this problem 😞 This issue is related to the [bug](https://github.com/meilisearch/meilisearch/issues/3021) introduced in v0.29, and you are experiencing this because your database is corrupted. To resolve this issue, please update to the latest version and remember to reindex your data to Meilisearch from scratch because using a dump will not work.

Again we are very sorry for the inconvenience, and thank you for using Meilisearch 🙂

**Comment by flexchar** [comment]
Thanks for clarifying @oluademola. If I understand right, this has been fully fixed in the latest version (as of today - 0.30.4)? If so, feel free to close the ticket. I'll update the instances asap. 

Appreciate you work, keep building it!

**Comment by flexchar** [comment]
Updated our instances to the latest v0.30.4 and damn me... indexing time from hours (~5h) to a minute or two. What a great Xmas treat! Thank you.
